arrows/mvg/camera_options.cxx

```diff
@@ -0,0 +1,206 @@
+  template <>                                                             \
+  NS::ceres_type                                                          \
+  config_block_get_value_cast( config_block_value_t const & value )       \
+  {                                                                       \
+    NS::ceres_type cet;                                                   \
+    if( !NS::StringTo##ceres_type( value, &cet ) )                        \
+    {                                                                     \
+      VITAL_THROW( bad_config_block_cast, value );                        \
+    }                                                                     \
+    return cet;                                                           \
+  }                                                                       \
+                                                                          \
+  }                                                                       \
+                                                                          \
+  namespace arrows {                                                      \
+  namespace mvg {                                                         \
+                                                                          \
+  template <>                                                             \
+  std::string                                                             \
+  mvg_options< NS::ceres_type >()                                         \
+  {                                                                       \
+    typedef NS::ceres_type T;                                             \
+    std::string options_str = "\nMust be one of the following options:";  \
+    std::string opt;                                                      \
+    for( unsigned i = 0; i < 20; ++i )                                    \
+    {                                                                     \
+      opt = NS::ceres_type##ToString( static_cast< T >( i ) );            \
+      if( opt == "UNKNOWN" )                                              \
+      {                                                                   \
+        break;                                                            \
+      }                                                                   \
+      options_str += "\n  - " + opt;                                      \
+    }                                                                     \
+    return options_str;                                                   \
+  }                                                                       \
+                                                                          \
+  }                                                                       \
+  }                                                                       \
+  }
+
+MVG_ENUM_HELPERS( kwiver::arrows::mvg, LensDistortionType )
```

Comment:
If this is only used once, *not* making it a macro may be preferred.

Reply:
I would keep this macro to be analogous to the ceres sources, and just in case we'd have more than one enum